### ISSUE TITLE
Compress messages by default

### ISSUE THREAD

**Issue by toberndo** [openpgpjs/openpgpjs#87]
openpgp.write_encrypted_message currently does not compress the message before encryption. This should be changed.

RFC 4880 says:

```
OpenPGP implementations SHOULD compress the message after applying
the signature but before encryption.
If an implementation does not implement compression, its authors should
be aware that most OpenPGP messages in the world are compressed.
...
Furthermore, compression has the added side effect that some types of 
attacks can be thwarted by the fact that slightly altered, compressed
data rarely uncompresses without severe errors.
```

Attacks on non-compressed messages: http://www.cs.umd.edu/~jkatz/papers/pgp-attack.pdf
##

**Comment by tanx** [comment]
Is this issue still relevant?

**Comment by laktak** [comment]
Compression by default would be nice. Also see https://github.com/openpgpjs/openpgpjs/pull/272#issuecomment-75221078

**Comment by bartbutler** [comment]
It would be. Last time I checked it was ridiculously slow, but we can
probably look for new ASM or other fast libraries to use for it just as we
did for AES.

**Comment by laktak** [comment]
> Last time I checked it was ridiculously slow

Did you check after #272?

**Comment by bartbutler** [comment]
I think I did. As another comment in that thread says, it doesn't make much sense to use JS when node and browser-native implementations exist.

I doubt compression is an issue for messages, it just doesn't make much difference in that case. For files I remember it being awful, though it should be revisited.

**Comment by chesnokovilya** [comment]
I think this issue  should be closed by PR  #637
